mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -2333,49 +2386,44 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
           targetGemmDim1Attr.push_back(
               b.getNamedAttr("upper_layer_names",
                              b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
-        } else if (arg1TargetLayoutName2 == "gemmK") {
+
+          if (convOpType == miopen::ConvOpType::Conv2DOpType) {
+            inputOobCheckDims.insert(nameToDims["ci"]);
+          } else if (convOpType == miopen::ConvOpType::Conv2DBwdWeightOpType) {
+            inputOobCheckDims.insert(nameToDims["ni"]);
+          }
+
+          inputOobCheckDims.insert(nameToDims["hi"]);
+          inputOobCheckDims.insert(nameToDims["wi"]);
+        }
+      }
+
+      if (inputCheckPadGemmN) {
+        if (arg1TargetLayoutName2 == "gemmN") {
           isInputPad = true;
           isGemmDim2Pad = true;
-          gemmDim2TargetName = b.getStringAttr(gemmKPad_name);
+          gemmDim2TargetName = b.getStringAttr(gemmNPad_name);
+          paddingInputShape[2] = paddingInputShape[2] + gemmNExtra;
 
-          paddingInputShape[2] = paddingInputShape[2] + gemmKExtra;
+          paddingInputShape[2] = paddingInputShape[2] + gemmNExtra;
           sourceGemmDim2Attr.push_back(
               b.getNamedAttr("transformation", b.getStringAttr("Pad")));
           sourceGemmDim2Attr.push_back(b.getNamedAttr(
               "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
-                                            b.getI32IntegerAttr(gemmKExtra)})));
-
+                                            b.getI32IntegerAttr(gemmNExtra)})));
           targetGemmDim2Attr.push_back(
               b.getNamedAttr("upper_layer_names",
-                             b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
-        }
-        // input of forward, gemmK = ci * y * x( y x from hi wi)
-        inputOobCheckDims.insert(nameToDims["ci"]);
-        inputOobCheckDims.insert(nameToDims["hi"]);
-        inputOobCheckDims.insert(nameToDims["wi"]);
-      }
-
-      if (gemmMExtra > 0) {
-        if (arg1TargetLayoutName1 == "gemmM") {
-          isInputPad = false;
-          isGemmDim1Pad = false;
-          paddingInputShape[1] = paddingInputShape[1] + gemmMExtra;
-        } else if (arg1TargetLayoutName2 == "gemmM") {
-          isInputPad = false;
-          isGemmDim2Pad = false;
-          paddingInputShape[2] = paddingInputShape[2] + gemmMExtra;
-        }
-      }
-
-      if (gemmNExtra > 0) {
-        if (arg1TargetLayoutName1 == "gemmN") {
-          isInputPad = false;
-          isGemmDim1Pad = false;
-          paddingInputShape[1] = paddingInputShape[1] + gemmNExtra;
-        } else if (arg1TargetLayoutName2 == "gemmN") {
-          isInputPad = false;
-          isGemmDim2Pad = false;
-          paddingInputShape[2] = paddingInputShape[2] + gemmNExtra;
+                             b.getArrayAttr({b.getStringAttr(gemmNPad_name)})));
+
+          if (convOpType == miopen::ConvOpType::Conv2DOpType) {
```

Comment:
please add comments here on the rationale assigning dimensions to OOB check dimensions, like what you did above for filter.

Reply:
unresolve the conversation as comments added above were not comprehensible